engine/execution/computation/committer/committer.go

```diff
@@ -58,10 +58,11 @@ func (s *LedgerViewCommitter) commitView(view state.View, baseState flow.StateCo
 }
 
 func (s *LedgerViewCommitter) collectProofs(view state.View, baseState flow.StateCommitment) (proof []byte, err error) {
+	// get all deduplicated register IDs
 	allIds := view.AllRegisters()
-	keys := make([]ledger.Key, len(allIds))
-	for i, id := range allIds {
-		keys[i] = execState.RegisterIDToKey(id)
+	keys := make([]ledger.Key, 0, len(allIds))
+	for _, id := range allIds {
+		keys = append(keys, execState.RegisterIDToKey(id))
```

Comment:
just out of curiosity, is it the case that the second version is faster because of random access?

Reply:
The second version is faster because it avoids initializing the array.  
`make([]int, 0, n)` allocates memory without initializing the array, while `make([]int, n)` allocates the memory and initializes it with a zero value.